src/app/core/services/phase.service.ts

```diff
@@ -200,16 +204,22 @@ export class PhaseService {
           return this.verifySessionAvailability(this.sessionData);
         }
       }),
-      flatMap((isSessionCreated: boolean) => {
-        if (!isSessionCreated) {
-          throw new Error('Session Availability Fix failed.');
-        }
-        return this.labelService.synchronizeRemoteLabels();
-      }),
+      this.checkSessionCreation(),
+      this.labelService.syncLabels(),
       retry(1)  // Retry once, to handle edge case where GitHub API cannot immediately confirm existence of the newly created repo.
     );
   }
 
+  public checkSessionCreation(): UnaryFunction<Observable<boolean>, Observable<boolean>> {
```

Comment:
I think we can just use `throwIfFalse` directly in the session setup pipeline

Reply:
👍